Recent Crash - Failed to connect during sensor update

<!-- READ THIS FIRST:
- Make sure you run the latest version of the Android app
- Make sure you run the latest version of Home Assistant
- Make sure to check the Companion docs for troubleshooting and configuration: https://companion.home-assistant.io/
- Make sure the bug you found is not already reported, we love to put work in bugfixes instead of closing duplicate bug reports
  DO NOT DELETE ANY TEXT from this template! All requested information is important.
-->

**Home Assistant Android version:**

latest master

**Android version:**

13

**Phone model:**

Pixel 7 Pro

**Home Assistant version:**

2022.12.x

**Last working Home Assistant release (if known):**

N/A

**Description of problem:**

Noticed the app crashed yesterday with the below error.

**Traceback (if applicable, to get the logs you may refer to: https://companion.home-assistant.io/docs/troubleshooting/faqs/#android-crash-logs):**

```
2022-12-13 15:59:21.177: io.homeassistant.companion.android.common.data.integration.IntegrationException: java.net.SocketTimeoutException: failed to connect to [DOMAIN/IP](http://DOMAIN/IP) (port 8123) from /[IP](http://IP/) (port 59744) after 10000ms
	at io.homeassistant.companion.android.common.data.integration.impl.IntegrationRepositoryImpl.updateSensors(IntegrationRepositoryImpl.kt:727)
	at io.homeassistant.companion.android.common.data.integration.impl.IntegrationRepositoryImpl$updateSensors$1.invokeSuspend(Unknown Source:15)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)	
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@faa7f96, Dispatchers.IO]Caused by: java.net.SocketTimeoutException: failed to connect to [DOMAIN/IP](http://DOMAIN/IP) (port 8123) from /[IP](http://IP/) (port 59744) after 10000ms
	at libcore.io.IoBridge.connectErrno(IoBridge.java:235
)	at libcore.io.IoBridge.connect(IoBridge.java:179)
	at java.net.PlainSocketImpl.socketConnect(PlainSocketImpl.java:142)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:390)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:230)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:212)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:436)
	at java.net.Socket.connect(Socket.java:646)
	at okhttp3.internal.platform.Platform.connectSocket(Platform.kt:128)
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:295)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207)
	at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226)
	at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106)
	at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74)
	at okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)	
	at java.lang.Thread.run(Thread.java:1012)
```

**Screenshot of problem:**

**Additional information:**
My HA instance was not down at this time but I still noticed this crash.

--- a/common/src/main/java/io/homeassistant/companion/android/common/sensors/SensorReceiverBase.kt
+++ b/common/src/main/java/io/homeassistant/companion/android/common/sensors/SensorReceiverBase.kt
@@ -338,9 +338,15 @@ abstract class SensorReceiverBase : BroadcastReceiver() {
         context: Context,
         sensorId: String
     ) {
-        val sensorManager = managers.firstOrNull { it.getAvailableSensors(context).any { s -> s.id == sensorId } }
-        sensorManager?.requestSensorUpdate(context)
-        val basicSensor = sensorManager?.getAvailableSensors(context)?.firstOrNull { it.id == sensorId }
+        val sensorManager = managers.firstOrNull {
+            it.getAvailableSensors(context).any { s -> s.id == sensorId }
+        } ?: return
+        try {
+            sensorManager.requestSensorUpdate(context)
+        } catch (e: Exception) {
+            Log.e(tag, "Issue requesting updates for ${context.getString(sensorManager.name)}", e)
+        }
+        val basicSensor = sensorManager.getAvailableSensors(context).firstOrNull { it.id == sensorId }
         val fullSensor = sensorDao.getFull(sensorId)
         if (
             fullSensor != null && fullSensor.sensor.enabled &&
@@ -350,12 +356,16 @@ abstract class SensorReceiverBase : BroadcastReceiver() {
                     fullSensor.sensor.icon != fullSensor.sensor.lastSentIcon
                 )
         ) {
-            integrationUseCase.updateSensors(arrayOf(fullSensor.toSensorRegistration(basicSensor)))
-            sensorDao.updateLastSentStateAndIcon(
-                basicSensor.id,
-                fullSensor.sensor.state,
-                fullSensor.sensor.icon
-            )
+            try {
+                integrationUseCase.updateSensors(arrayOf(fullSensor.toSensorRegistration(basicSensor)))
+                sensorDao.updateLastSentStateAndIcon(
+                    basicSensor.id,
+                    fullSensor.sensor.state,
+                    fullSensor.sensor.icon
+                )
+            } catch (e: Exception) {
+                Log.e(tag, "Exception while updating individual sensor.", e)
+            }
         }
     }
 

